src/core/reference/include/openvino/reference/erf.hpp

```diff
@@ -6,22 +6,34 @@
 
 #include <cmath>
 #include <cstddef>
-#include <type_traits>
+
+#include "openvino/reference/utils/type_util.hpp"
 
 namespace ov {
 namespace reference {
-template <typename T, typename std::enable_if<!std::is_integral<T>::value, bool>::type = true>
-void erf(const T* arg, T* out, size_t count) {
-    for (size_t i = 0; i < count; i++) {
-        out[i] = static_cast<T>(std::erf(arg[i]));
-    }
+namespace func {
+
+template <class T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+T erf(const T v) {
+    return static_cast<T>(std::round(std::erf(v)));
+}
+
+template <class T, typename std::enable_if<ov::is_floating_point<T>()>::type* = nullptr>
+T erf(const T v) {
+    return std::erf(v);
```

Comment:
`static_cast<T>` is missed - may be highlighted by static analysis like `coverity`

Reply:
I thin it should not be issue here.
The `float16`, `bfloat16` have defined conversion from/to `float`, but if you are sure that there will be an issue reported by `Coverity` tool, I can add cast. 
I'd like avoid to add it `just-in-case`.

Reply:
I'm not 100% sure, so final version is up to you